src/osal/osal_none.h

```diff
@@ -203,6 +202,15 @@ static inline bool osal_queue_send(osal_queue_t const qhdl, void const * data, b
   return success;
 }
 
+static inline bool osal_queue_empty(osal_queue_t qhdl)
+{
+  _osal_q_lock(qhdl);
```

Comment:
This approach won't work.

The user needs to call `tud_task_event_ready()` with interrupts already disabled in order to check whether the queue is empty. If the queue is empty, then it can issue `WFI`. If not, it will re-enable interrupts and not issue `WFI`.

I'm uncertain what happens if you issue `cpsid i` when interrupts are already disabled, but this function will re-enable them when it exits. If a USB event comes in between the time this function exits and when the calling function issues `WFI`, then that interrupt will be handled and the queue will no longer be empty, but the calling function will still issue `WFI`.

This function needs to be called with interrupts already disabled, and shouldn't lock/unlock the system on its own.

Reply:
@xobs Yeah, you are right, I was thinking to have this function being used in normal/general mode as well. However, there is not much benefit in normal mode, also locking the queue is not crucial as well in that mode.
  
NVIC enable probably shouldn't have any impact on the global CPSID, but I could be wrong. I   will remove the queue lock/unlock anyway since it is better to do.